File: mycuration/src/main/java/com/phicdy/mycuration/view/AddCurationView.java
package com.phicdy.mycuration.view;

import java.util.ArrayList;

public interface AddCurationView {
    int editCurationId();
    String inputWord();
    String curationName();
    ArrayList<String> wordList();
    void setCurationName(String name);
    void resetInputWord();
    void refreshList();
    void addWord(String word);
    void setWords(ArrayList<String> words);
    void setTitleForEdit();
    void handleEmptyCurationNameError();
    void handleEmptyWordError();
    void handleSameNameCurationError();
    void handleAddSuccess();
    void handleEditSuccess();
    void showSuccessToast();
    void showErrorToast();
    void showWordEmptyErrorToast();
    void showToast(String text);
    void showProgressDialog();
    void dismissProgressDialog();
    void finish();
}

File: legacy/src/main/java/com/phicdy/mycuration/presentation/view/AddCurationView.kt
package com.phicdy.mycuration.presentation.view

interface AddCurationView {
    fun curationName(): String
    fun resetInputWord()
    fun handleEmptyCurationNameError()
    fun handleEmptyWordError()
    fun handleSameNameCurationError()
    fun handleAddSuccess()
    fun handleEditSuccess()
    fun showSuccessToast()
    fun showErrorToast()
    fun showToast(text: String)
    fun showProgressDialog()
    fun dismissProgressDialog()
    fun finish()
}
